pkg/mcp/source/server_source.go

```diff
@@ -52,27 +52,26 @@ var _ mcp.ResourceSourceServer = &Server{}
 // ServerOptions contains sink server specific options
 type ServerOptions struct {
 	AuthChecker AuthChecker
-	RateLimiter internal.RateLimit
+	RateLimiter rate.Limit
 }
 
 // NewServer creates a new instance of a MCP source server.
 func NewServer(srcOptions *Options, serverOptions *ServerOptions) *Server {
-	rateLimiter := serverOptions.RateLimiter
-	if rateLimiter == nil {
-		rateLimiter = internal.NewNoopRateLimiter()
-	}
 	s := &Server{
 		src:         New(srcOptions),
 		authCheck:   serverOptions.AuthChecker,
-		rateLimiter: rateLimiter,
+		rateLimiter: serverOptions.RateLimiter,
 	}
 	return s
 }
 
 // EstablishResourceStream implements the ResourceSourceServer interface.
 func (s *Server) EstablishResourceStream(stream mcp.ResourceSource_EstablishResourceStreamServer) error {
-	if err := s.rateLimiter.Wait(stream.Context()); err != nil {
-		return err
+	if s.rateLimiter != nil {
```

Comment:
Do we need this after @ozevren added the noop rate limiter?

Reply:
Had a conversation with @ozevren earlier this morning, the noop rate limiter is removed in favour of the if block above however when I pick up the GH issue to make ratelimiters configurable I will be adding a proper validation for `srcOptions` and `serverOptions` which would align more with other patterns we have in other components.